File: java/postpony-api-sdk/src/test/java/com/postpony/PostponyApiServiceTest.java
package com.postpony;


import com.postpony.common.ClientManager;
import com.postpony.common.JabxXmlUtil;
import com.postpony.model.*;

import java.io.BufferedOutputStream;
import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.math.BigDecimal;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

class PostponyApiServiceTest {

    public PostponyApiServiceTest()
    {
        //账号信息可通过postpony客服或技术获取
        ClientManager.SetCurrent("","","");
    }

    @org.junit.jupiter.api.Test
    void getRate() {
        try{
            RateRequest request=createRateRequest();

            RateResponse response= PostponyApiService.getRate(request);
            if(response.getFedex().isSucess()==true)
            {
                System.out.println("fedex获取报价成功");
            }
            if(response.getUsps().isSucess()==true)
            {
                System.out.println("usps获取报价成功");
            }


            if(response.getUps().isSucess()==true)
            {
                System.out.println("ups获取报价成功");
            }

            if(response.getDhl().isSucess()==true)
            {
                System.out.println("dhl获取报价成功");
            }
        }catch (Exception ex){

        }
        //ClientManager.SetCurrent("PY","pypypypypy","https://apitest.postpony.com/");

    }

    @org.junit.jupiter.api.Test
    void ship() {

        ShipRequest request=CreateShipRequest();
        ShipResponse response= PostponyApiService.ship(request);
        if(response.isSucess()){
            System.out.println("生成面单成功");
            CreateFile(response.getLableDataList().get(0));
        }
    }

    @org.junit.jupiter.api.Test
    void getTracking() {
        TrackingRequest request = new TrackingRequest();
        request.setLogisticsCode(ShippingClassEnum.Usps.toString());
        request.setTrackNo("9400110205425007474346");
        TrackingResponse result = PostponyApiService.getTracking(request);
        if(result.isSucess()){
            System.out.println("物流跟踪信息获取成功");
        }
    }

    @org.junit.jupiter.api.Test
    void downloading() {
        DownloadRequest request = new DownloadRequest();
        request.setLabelFormatType("PDF");
        request.setLabelId(768208);
        DownLoadRespose result = PostponyApiService.downloading(request);
        if(result.isSucess()){
            System.out.println("面单下载成功");
        }
    }

    @org.junit.jupiter.api.Test
    void getLabelInfo() {
        LabelIdRequest request = new LabelIdRequest();
        request.setOrderId("dk8391kdk28dk3dddd902k2");
        LabelIdRespose result = PostponyApiService.getLabelInfo(request);
        if(result.isSucess()){
            System.out.println("根据订单号获取label信息成功");
        }
    }

    @org.junit.jupiter.api.Test
    void batchDownload() {
        BatchDownloadRequest request = new BatchDownloadRequest();
        request.setLabelIds(new ArrayList<Integer>());
        request.getLabelIds().add(768208);
        request.getLabelIds().add(768209);
        BatchDownLoadRespose result = PostponyApiService.batchDownload(request);
        if(result.isSucess())
        {
            System.out.println("批量生成面单成功，面单下载地址："+result.getPdfUrl());
        }
    }

    @org.junit.jupiter.api.Test
    void getShipDetail() {
        ShippingDetailRequest request = new ShippingDetailRequest();
        request.setLabelId(768208);

        ShippingDetailResponse result = PostponyApiService.getShipDetail(request);
        if(result.isSucess()){
            System.out.println("label信息查询成功");
        }
    }

    private RateRequest createRateRequest()
    {
        RateRequest rateRequest = new RateRequest();
        rateRequest.setCanGetBaseRate(false);
        rateRequest.setOriginalAddress(new AddressInfo());
        rateRequest.getOriginalAddress().setPersonName("test");
        rateRequest.getOriginalAddress().setCity("Portland");
        rateRequest.getOriginalAddress().setPostalCode("97230");
                rateRequest.getOriginalAddress().setStateOrProvinceCode("OR");
                rateRequest.getOriginalAddress().setCountryCode("US");
                rateRequest.getOriginalAddress().setPhoneNumber("2025551212");
                rateRequest.getOriginalAddress().setStreetLines(new String[]{ "14121 NE Airport Way" });
        rateRequest.setDestinationAddress(new AddressInfo());
        rateRequest.getDestinationAddress().setPersonName( "test");
        rateRequest.getDestinationAddress().setCity( "Portland");
        rateRequest.getDestinationAddress().setPostalCode( "97230");
        rateRequest.getDestinationAddress().setStateOrProvinceCode( "OR");
        rateRequest.getDestinationAddress().setCountryCode( "US");

        rateRequest.getDestinationAddress().setPhoneNumber( "2025551212");
        rateRequest.getDestinationAddress().setStreetLines( new String [] { "14121 NE Airport Way" });

        rateRequest.setPackageInfos(new ArrayList<PackageItemInfo>());
        PackageItemInfo packageitem= new PackageItemInfo();
        packageitem.setHeight(new BigDecimal(1));
        packageitem.setWeight(new BigDecimal(1));
        packageitem.setWidth(new BigDecimal(1));
        packageitem.setLength(new BigDecimal(1));
        rateRequest.getPackageInfos().add(packageitem);

        rateRequest.setUpsRate(true);
        return rateRequest;
    }

    private static ShipRequest CreateShipRequest()
    {
        ShipRequest shipRequest = new ShipRequest();
        ShipRequestDetail requstInfo=new ShipRequestDetail();
        AddressInfo shipper = new AddressInfo();
        shipper.setPersonName("test");
        shipper.setCity("Portland");
        shipper.setStateOrProvinceCode("OR");
        shipper.setPostalCode("97230");
        shipper.setCountryCode("US");
        shipper.setPhoneNumber("2025551212");
        shipper.setStreetLines(new String[]{"14121 NE Airport Way"});
        requstInfo.setShipper(shipper);

        AddressInfo recipient = new AddressInfo();
        recipient.setPersonName("test");
        recipient.setCity("Portland");
        recipient.setStateOrProvinceCode("OR");
        recipient.setPostalCode("97230");
        recipient.setCountryCode("US");
        recipient.setPhoneNumber("2025551212");
        recipient.setStreetLines(new String[]{"14121 NE Airport Way"});
        requstInfo.setRecipient(recipient);

        List<PackageItemInfo> packageItemInfos=new ArrayList<PackageItemInfo>();

        for (int i=0;i<2;i++){
            PackageItemInfo item=new PackageItemInfo();
            item.setWidth(new BigDecimal(1));
            item.setHeight(new BigDecimal(1));
            item.setLength(new BigDecimal(1));
            item.setWeight(new BigDecimal(1));
            packageItemInfos.add(item);
        }
        requstInfo.setPackageItems(packageItemInfos);
        requstInfo.setPackageItems(packageItemInfos);
        shipRequest.setAuthorizedKey("TESTTOKEN-ske39De3mkC39d");
        shipRequest.setLabelFormatType("PDF");
        shipRequest.setNoBackData(false);
        //orderid 是用于标识提交的订单的唯一性，会打印到面单上，orderid可为空
        //  shipRequest.OrderId = "9387291183320111";


        shipRequest.setShipType(ShipTypeEnum.UpsGround.toString());


        requstInfo.setLbSize(LabelSize.S4X6.toString());
        WaybillInformation waybillInformation=new WaybillInformation();
        Date date=new Date();
        SimpleDateFormat dateFormat=new SimpleDateFormat("yyyy-MM-dd");
        waybillInformation.shipDate = dateFormat.format(date);
        requstInfo.setPackageInfo(waybillInformation);
        //备注，会打印到面单上
        waybillInformation.setShippingNotes("testsetsetsetsteste");
        requstInfo.setPackageInfo(waybillInformation);
        shipRequest.setRequstInfo(requstInfo);
        shipRequest.setOrderId("dk8391kdk22dddd902k2");
        return shipRequest;
    }

    private void CreateFile(byte[] data)
    {
        try{
            String filePath="/Users/liyoujin/testpdf/javatest.pdf";
            BufferedOutputStream bos = null;
            FileOutputStream fos = null;
            File file = null;
            try {
                file = new File(filePath);
                fos = new FileOutputStream(file);
                bos = new BufferedOutputStream(fos);
                bos.write(data);
            } catch (Exception e) {
                e.printStackTrace();
            } finally {
                if (bos != null) {
                    try {
                        bos.close();
                    } catch (IOException e1) {
                        e1.printStackTrace();
                    }
                }
                if (fos != null) {
                    try {
                        fos.close();
                    } catch (IOException e1) {
                        e1.printStackTrace();
                    }
                }
            }
        }catch (Exception ex)
        {

        }

    }
}

File: dotnet/postpony.api.sdk/Postpony.Api.Sdk.Test/PostponyApiServiceTest.cs
using Postpony.Api.Sdk;
using System;
using System.Collections.Generic;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Postpony.Api.Sdk.Common;
using Postpony.Api.Sdk.Enum;
using Postpony.Api.Sdk.Model;

namespace Postpony.Api.Sdk.Test
{



    [TestClass()]
    public class PostponyApiServiceTest
    {

        [TestInitialize]
        public void Initialize()
        {
            //todo 账号信息需要通过postpony 客服或者 技术 获得。
            ClientManager.SetCurrent("", "", "");
        }

        [TestMethod()]
        public void GetRateTest()
        {
            try
            {
                var request = CreateRateRequest();
                var result = PostponyApiService.GetRate(request);
                Assert.AreEqual(result.Sucess, true);
                Assert.AreEqual(result.Fedex.Sucess, true);
                Assert.AreEqual(result.Ups.Sucess, true);
                Assert.AreEqual(result.Usps.Sucess, true);
                Assert.AreEqual(result.Dhl.Sucess, true);
            }
            catch (Exception ex)
            {

            }

        }

        [TestMethod()]
        public void ShipTest()
        {
            try
            {
                var result = PostponyApiService.Ship(CreateShipRequest());
                Assert.AreEqual(result.Sucess, true);
            }
            catch (Exception)
            {

                throw;
            }

        }

        private static ShipRequest CreateShipRequest()
        {
            ShipRequest shipRequest = new ShipRequest();
            shipRequest.RequstInfo = new ShipRequestDetail();
            shipRequest.RequstInfo.Shipper = new AddressInfo()
            {
                PersonName = "test",
                City = "Portland",
                PostalCode = "97230",
                StateOrProvinceCode = "OR",
                CountryCode = "US",
                StreetLines = new[] {"14121 NE Airport Way"},
                PhoneNumber = "2025551212",
            };
            shipRequest.RequstInfo.Recipient = new AddressInfo()
            {
                PersonName = "test",
                City = "Portland",
                PostalCode = "97230",
                StateOrProvinceCode = "OR",
                CountryCode = "US",
                StreetLines = new[] {"14121 NE Airport Way"},
                PhoneNumber = "2025551212",
            };

            shipRequest.RequstInfo.PackageItems = new List<PackageItemInfo>()
            {
                new PackageItemInfo() {Width = 1, Height = 1, Length = 1, Weight = 1}
            };
            shipRequest.AuthorizedKey = "TESTTOKEN-ske39De3mkC39d";
            shipRequest.LabelFormatType = "PDF";
            shipRequest.NoBackData = true;
            //orderid 是用于标识提交的订单的唯一性，会打印到面单上，orderid可为空
            // shipRequest.OrderId = "6666666666";


            shipRequest.ShipType = ShipTypeEnum.FedEx2Day.ToString();


            shipRequest.RequstInfo.LbSize = LabelSize.S4X6.ToString();
            shipRequest.RequstInfo.Package = new WaybillInformation();
            shipRequest.RequstInfo.Package.ShipDate = DateTime.Now.AddDays(2);

            //备注，会打印到面单上
            shipRequest.RequstInfo.Package.ShippingNotes = "testsetsetsetsteste";
            return shipRequest;
        }

        [TestMethod()]
        public void GetTrackingTest()
        {
            try
            {
                var request = new TrackingRequest()
                {
                    TrackNo = "9400110205425007474346",
                    LogisticsCode = ShippingClassEnum.Usps.ToString()
                };
                var result = PostponyApiService.GetTracking(request);
                Assert.AreEqual(result.Sucess, true);
            }
            catch (Exception ex)
            {
                throw;
            }
        }

        [TestMethod()]
        public void DownloadingTest()
        {
            try
            {
                var request = new DownloadRequest() {LabelId = 758631, LabelFormatType = LabelFormatEnum.PDF.ToString()};
                var result = PostponyApiService.Downloading(request);
                Assert.AreEqual(result.Sucess, true);
            }
            catch (Exception ex)
            {
                throw;
            }

        }

        [TestMethod()]
        public void GetLabelInfoTest()
        {
            try
            {
                var request = new LabelIdRequest() {OrderId = "L-21-2560666"};
                var result = PostponyApiService.GetLabelInfo(request);
                Assert.AreEqual(result.Sucess, true);
            }
            catch (Exception ex)
            {
                throw;
            }
        }

        [TestMethod()]
        public void BatchDownloadTest()
        {
            try
            {
                var request = new BatchDownloadRequest() { LabelIds = new List<int>() { 758631, 758645 } };
                var result = PostponyApiService.BatchDownload(request);
                Assert.AreEqual(result.Sucess, true);
            }
            catch (Exception ex)
            {
                throw;
            }
        }

        private RateRequest CreateRateRequest()
        {
            RateRequest rateRequest = new RateRequest();
            rateRequest.CanGetBaseRate = false;
            rateRequest.OriginalAddress = new AddressInfo()
            {
                PersonName = "test",
                City = "Portland",
                PostalCode = "97230",
                StateOrProvinceCode = "OR",
                CountryCode = "US",
                PhoneNumber = "2025551212",
                StreetLines = new[] { "14121 NE Airport Way" },
            };
            rateRequest.DestinationAddress = new AddressInfo()
            {
                PersonName = "test",
                City = "Portland",
                PostalCode = "97230",
                StateOrProvinceCode = "OR",
                CountryCode = "US",

                PhoneNumber = "2025551212",
                StreetLines = new[] { "14121 NE Airport Way" },
            };

            rateRequest.PackageInfos = new List<PackageItemInfo>()
            {
                new PackageItemInfo() {Width=1,Height = 1,Length = 1,Weight = 1}
            };
            rateRequest.UpsRate = true;
            return rateRequest;
        }

        [TestMethod()]
        public void GetShipDetailTest()
        {
            var request = new ShippingDetailRequest() { LabelId = 765361};
            var result = PostponyApiService.GetShipDetail(request);
            Assert.AreEqual(result.Sucess, true);
        }
    }
}